File: react-native/android/src/main/java/com/ricoh360/thetableclientreactnative/WlanControlCommandService.kt
package com.ricoh360.thetableclientreactnative

import com.facebook.react.bridge.Promise

object WlanControlCommandService {
  suspend fun getWlanPasswordState(id: Int, promise: Promise) {
    try {
      val device = ThetaBleClientReactNativeModule.deviceList[id]
      device ?: let {
        promise.reject(Exception(ERROR_MESSAGE_DEVICE_NOT_FOUND))
        return
      }
      val service = device.wlanControlCommand
      service ?: let {
        promise.reject(Exception(ERROR_MESSAGE_UNSUPPORTED_SERVICE))
        return
      }
      val value = service.getWlanPasswordState()
      promise.resolve(value.name)
    } catch (e: Throwable) {
      promise.reject(e)
    }
  }
}

File: react-native/ios/WlanControlCommandService.swift
//
//  WlanControlCommandService.swift
//  ThetaBleClientReactNative
//
//  Created on 2025/06/20.
//

import THETABleClient

enum WlanControlCommandService {
    static func getWlanPasswordState(id: Int,
                                     resolve: @escaping RCTPromiseResolveBlock,
                                     reject: @escaping RCTPromiseRejectBlock)
    {
        guard let device = ThetaBleClientReactNative.deviceList[id] else {
            reject(ERROR_TITLE, ERROR_MESSAGE_DEVICE_NOT_FOUND, nil)
            return
        }
        guard let service = device.wlanControlCommand else {
            reject(ERROR_TITLE, ERROR_MESSAGE_UNSUPPORTED_SERVICE, nil)
            return
        }

        Task {
            do {
                let value = try await service.getWlanPasswordState()
                resolve(value.name)
            } catch {
                reject(ERROR_TITLE, error.localizedDescription, error)
            }
        }
    }
}
